src/components/games/audio-challenge/audio-challenge.tsx

```diff
@@ -0,0 +1,390 @@
+import React, { useState, useEffect, useRef, useMemo } from 'react';
+import { WordsProps, GameProps, StatisticsProps } from "../../../common/ts/interfaces";
+import { NavLink } from 'react-router-dom';
+import { connect } from "react-redux";
+import { Loader } from "../../loader";
+import { ResultsGame } from '../resultsGame/resultsGame';
+import { Lives } from "../lives/lives";
+import { Sound } from "../sound";
+import {
+  shuffleArray, highlightWords, removeWordsHighlighting,
+  onFullScreenClick, changeFullscreenIcon
+} from "../utils/utils";
+import { playAnswer, updateStatistics } from '../../../data/utils';
+import { urlBackend } from "../../../data";
+import { setData, getStatistics, setStatistics } from '../../../data';
+
+let necessaryWords: WordsProps[];
+let correctAnswers: WordsProps[] = [];
+let wrongAnswers: WordsProps[] = [];
+let lives: number = 5;
+let answers: number = 0;
+let isMute: boolean = false;
+let audio = new Audio();
+let statisticBack: StatisticsProps;
+let series: number = 0;
+let seriesMax: number = 0;
+
+interface SavannahProps {
+  words: WordsProps[],
+  user: {},
+};
+
+const AudioChallengeRedux: React.FC<GameProps & SavannahProps> = ({ group, page = -1, words, user }) => {
+  const [gameWords, setGameWords] = useState<WordsProps[]>([]);
+  const [translations, setTranslations] = useState<WordsProps[]>([]);
+  const [gameType, setGameType] = useState<string>(``);
+  const [isWelcomeScreen, setIsWelcomeScreen] = useState<boolean>(true);
+  const audioChallenge = useRef<HTMLElement>(null);
+  const soundWaves = useRef<HTMLImageElement>(null);
+  const wordContainer = useRef<HTMLDivElement>(null);
+  const word = useRef<HTMLSpanElement>(null);
+  const button = useRef<HTMLButtonElement>(null);
+  const fullscreen = useRef<HTMLButtonElement>(null);
+
+  useEffect(() => {
+    page > -1 ?
+      necessaryWords = words.slice((Number(group) * 600), ((Number(group)) * 600) + ((page + 1) * 20)) :
+      necessaryWords = words.slice((Number(group) * 600), ((Number(group) + 1) * 600));
+
+    setGameWords(necessaryWords);
+    setTranslations(necessaryWords);
+
+    correctAnswers = [];
+    wrongAnswers = [];
+    answers = 0;
+    series = 0;
+    seriesMax = 0;
+    lives = 5;
+  }, [words, group, page]);
+
+  useEffect(() => {
+    getStatistics(user).then((res: any) => statisticBack = res.statistics);
+  }, []);
+
+  useEffect(() => {
+    if (lives === 0 || gameWords.length === 0) {
+      setGameStatistic();
+    }
+    return () => {
+      if (statisticBack) statisticBack = updateStatistics('audioCall', statisticBack, wrongAnswers, correctAnswers, seriesMax)
+      if (lives > 0 || gameWords.length === 0) setGameStatistic();
+    }
+  }, [lives, gameWords]);
+
+  useEffect(() => {
+    window.document.addEventListener("fullscreenchange", onFullScreenChange)
+    window.addEventListener("keyup", onKeyUpHandler);
+
+    return () => {
+      (window as any).removeEventListener("keyup", onKeyUpHandler);
```

Comment:
`window as Window` I think that can be done in that way. Any is something that we should avoid.